internal/fuzzbinder/fuzzclaim.go

```diff
@@ -180,3 +181,122 @@ func parseFuzzClaimBytes(statementBytes []byte) (*intoto.Statement, error) {
+	fuzzersFuzzEffort := make(map[string]*FuzzEffort)
+	fuzzersCoverage := make(map[string]*Coverage)
+	hyphenDate := fmt.Sprintf("%s-%s-%s", date[:4], date[4:6], date[6:])
+	//Get fuzzing statistics.
+	projectCoverage, err := GetCoverage(date, fuzzParameters, "perProject", "")
+	if err != nil {
+		return nil, err
+	}
+
+	for _, fuzzTarget := range fuzzTargets {
+		coverage, err := GetCoverage(date, fuzzParameters, "perTarget", fuzzTarget)
+		if err != nil {
+			return nil, err
+		}
+		fuzzEffort, err := GetFuzzEffort(revisionDigest, hyphenDate, fuzzTarget, fuzzParameters)
+		if err != nil {
+			return nil, err
+		}
+		crash, err := GetCrashes(revisionDigest, hyphenDate, fuzzTarget, fuzzParameters)
+		if err != nil {
+			return nil, err
+		}
+
+		fuzzersCrashes[fuzzTarget] = crash
+		fuzzersFuzzEffort[fuzzTarget] = fuzzEffort
+		fuzzersCoverage[fuzzTarget] = coverage
+
+		projectCrashes.Detected = projectCrashes.Detected || crash.Detected
+		projectFuzzEffort.FuzzTimeSeconds += fuzzEffort.FuzzTimeSeconds
+		projectFuzzEffort.NumberFuzzTests += fuzzEffort.NumberFuzzTests
+	}
+	// Generate fuzzing claim specification.
+	fuzzClaimSpec.PerProject = &FuzzStats{
+		BranchCoverage:  projectCoverage.BranchCoverage,
+		LineCoverage:    projectCoverage.LineCoverage,
+		DetectedCrashes: projectCrashes.Detected,
+		FuzzTimeSeconds: projectFuzzEffort.FuzzTimeSeconds,
+		NumberFuzzTests: projectFuzzEffort.NumberFuzzTests,
+	}
+	for _, fuzzTagret := range fuzzTargets {
+		targetStats := FuzzStats{
+			BranchCoverage:  fuzzersCoverage[fuzzTagret].BranchCoverage,
+			LineCoverage:    fuzzersCoverage[fuzzTagret].LineCoverage,
+			DetectedCrashes: fuzzersCrashes[fuzzTagret].Detected,
+			FuzzTimeSeconds: fuzzersFuzzEffort[fuzzTagret].FuzzTimeSeconds,
+			NumberFuzzTests: fuzzersFuzzEffort[fuzzTagret].NumberFuzzTests,
+		}
+		targetSpec := FuzzSpecPerTarget{
+			Name:      fmt.Sprintf("%s_%s_%s", fuzzParameters.FuzzEngine, fuzzParameters.ProjectName, fuzzTagret),
+			Path:      fmt.Sprintf("%s/fuzz/fuzz_targets/%s.rs", fuzzParameters.ProjectName, fuzzTagret),
+			FuzzStats: &targetStats,
+		}
+		fuzzClaimSpec.PerTarget = append(fuzzClaimSpec.PerTarget, targetSpec)
+	}
+	return &fuzzClaimSpec, nil
+}
+
+// GenerateFuzzClaim generates a fuzzing claim (an instance of intoto.Statement,
+// with AmberClaimV1 as the PredicateType and FuzzClaimV1 as the ClaimType) using the
+// fuzzing reports of OSS-Fuzz.
+func GenerateFuzzClaim(date string, fuzzParameters *FuzzParameters) (*intoto.Statement, error) {
+	var statement intoto.Statement
+	var predicate amber.ClaimPredicate
+	revisionDigest, err := GetCoverageRevision(date, fuzzParameters)
+	if err != nil {
+		return nil, err
+	}
+	fuzzTargets, err := GetFuzzTargets(date, fuzzParameters)
+	if err != nil {
+		return nil, err
+	}
+	// Generate Amber predicate
+	predicate.ClaimType = FuzzClaimV1
+	currentTime := time.Now()
+	tomorrow := time.Now().AddDate(0, 0, 1)
+	nextWeek := time.Now().AddDate(0, 0, 7)
```

Comment:
I guess we don't want these claims to be valid for only 7 days. I think the validity duration should be an input parameter. 

Feel free to add a TODO here and fix it in a future PR.

Reply:
I added a TODO: `TODO(#173): Add validity duration as an input parameter.`